### ISSUE TITLE
cannot install anymore

### ISSUE THREAD

**Issue by UnixJunkie** [BenevolentAI/guacamol_baselines#12]
```
$ pip3 install -r requirements.txt 
Collecting guacamol==0.5.3
  Using cached guacamol-0.5.3-py3-none-any.whl (37 kB)
Collecting matplotlib==3.0.2
  Using cached matplotlib-3.0.2.tar.gz (36.5 MB)
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python3 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-aa1wywkc/matplotlib/setup.py'"'"'; __file__='"'"'/tmp/pip-install-aa1wywkc/matplotlib/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-install-aa1wywkc/matplotlib/pip-egg-info
         cwd: /tmp/pip-install-aa1wywkc/matplotlib/
    Complete output (47 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-aa1wywkc/matplotlib/setup.py", line 170, in <module>
        msg = pkg.install_help_msg()
      File "/tmp/pip-install-aa1wywkc/matplotlib/setupext.py", line 539, in install_help_msg
        release = platform.linux_distribution()[0].lower()
    AttributeError: module 'platform' has no attribute 'linux_distribution'
    ============================================================================
    Edit setup.cfg to change the build options
    
    BUILDING MATPLOTLIB
                matplotlib: yes [3.0.2]
                    python: yes [3.8.10 (default, Sep 28 2021, 16:10:42)  [GCC
                            9.3.0]]
                  platform: yes [linux]
    
    REQUIRED DEPENDENCIES AND EXTENSIONS
                     numpy: yes [version 1.19.2]
          install_requires: yes [handled by setuptools]
                    libagg: yes [pkg-config information for 'libagg' could not
                            be found. Using local copy.]
                  freetype: no  [The C/C++ header for freetype2 (ft2build.h)
                            could not be found.  You may need to install the
                            development package.]
                       png: yes [version 1.6.37]
                     qhull: yes [pkg-config information for 'libqhull' could not
                            be found. Using local copy.]
    
    OPTIONAL SUBPACKAGES
               sample_data: yes [installing]
                  toolkits: yes [installing]
                     tests: no  [skipping due to configuration]
            toolkits_tests: no  [skipping due to configuration]
    
    OPTIONAL BACKEND EXTENSIONS
                       agg: yes [installing]
                     tkagg: yes [installing; run-time loading from Python Tcl /
                            Tk]
                    macosx: no  [Mac OS-X only]
                 windowing: no  [Microsoft Windows only]
    
    OPTIONAL PACKAGE DATA
                      dlls: no  [skipping due to configuration]
    
    ============================================================================
                            * The following required packages can not be built:
                            * freetype
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by UnixJunkie** [comment]
If I comment out matplotlib from the requirements.txt file, it goes further but still an error:
```
ERROR: Could not find a version that satisfies the requirement tensorflow==1.15.4 (from guacamol==0.5.3->-r requirements.txt (line 1)) (from versions: 2.2.0rc1, 2.2.0rc2, 2.2.0rc3, 2.2.0rc4, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.3.0rc0, 2.3.0rc1, 2.3.0rc2, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.4.0rc0, 2.4.0rc1, 2.4.0rc2, 2.4.0rc3, 2.4.0rc4, 2.4.0, 2.4.1, 2.4.2, 2.4.3, 2.4.4, 2.5.0rc0, 2.5.0rc1, 2.5.0rc2, 2.5.0rc3, 2.5.0, 2.5.1, 2.5.2, 2.6.0rc0, 2.6.0rc1, 2.6.0rc2, 2.6.0, 2.6.1, 2.6.2, 2.7.0rc0, 2.7.0rc1, 2.7.0)
ERROR: No matching distribution found for tensorflow==1.15.4 (from guacamol==0.5.3->-r requirements.txt (line 1))
```

**Comment by UnixJunkie** [comment]
Maybe the problem is that a compatible version of moses need to be specified in requirements.txt.

**Comment by UnixJunkie** [comment]
Maybe, the problem comes from guacamol:
```
# pip3 install guacamol
Collecting guacamol
  Using cached guacamol-0.5.3-py3-none-any.whl (37 kB)
ERROR: Could not find a version that satisfies the requirement tensorflow==1.15.4 (from guacamol) (from versions: 2.2.0rc1, 2.2.0rc2, 2.2.0rc3, 2.2.0rc4, 2.2.0, 2.2.1, 2.2.2, 2.2.3, 2.3.0rc0, 2.3.0rc1, 2.3.0rc2, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.4.0rc0, 2.4.0rc1, 2.4.0rc2, 2.4.0rc3, 2.4.0rc4, 2.4.0, 2.4.1, 2.4.2, 2.4.3, 2.4.4, 2.5.0rc0, 2.5.0rc1, 2.5.0rc2, 2.5.0rc3, 2.5.0, 2.5.1, 2.5.2, 2.6.0rc0, 2.6.0rc1, 2.6.0rc2, 2.6.0, 2.6.1, 2.6.2, 2.7.0rc0, 2.7.0rc1, 2.7.0)
ERROR: No matching distribution found for tensorflow==1.15.4 (from guacamol)
```

**Comment by UnixJunkie** [comment]
It would be nice if guacamol and guacamol_baselines can co-install.

**Comment by JoshuaMeyers** [comment]
Hey @UnixJunkie, apologies many of the authors have moved on and we haven't tried a fresh build of the two repos for a while. I'll take a look. In the meantime, the exact version of Moses needed pre-dates their PyPI releases:

`pip install git+https://github.com/molecularsets/moses.git@96c77d52c688dcac8e2962d78965175e7824c48c#egg=moses`

**Comment by JoshuaMeyers** [comment]
Regarding the tensorflow missing distribution, what version of Python are you using? This version of tensorflow is only available on Python 3.6-3.7

Link:
- [tensorflow system requirements](https://www.tensorflow.org/install/pip#system-requirements).

**Comment by JoshuaMeyers** [comment]
Hey @UnixJunkie, installation issues should be resolved by https://github.com/BenevolentAI/guacamol_baselines/pull/14. 

Going forward `guacamol_baselines` includes a Dockerfile for installing `guacamol_baselines` + `guacamol` together. When benchmarking your own models, you will only need to install `guacamol` from `pip` along with whatever dependencies are required for running your own model.
